sdks/java/core/src/test/java/org/apache/beam/sdk/util/ReduceFnRunnerTest.java

```diff
@@ -226,6 +227,33 @@ public void testOnElementCombiningDiscarding() throws Exception {
     tester.assertHasOnlyGlobalAndFinishedSetsFor(firstWindow);
   }
 
+  /**
+   * Tests that the garbage collection time for a fixed window does not overflow
+   * the end of time.
+   */
+  @Test
+  public void testFixedWindowEndOfTimeGarbageCollection() throws Exception {
+    // Test basic execution of a trigger using a non-combining window set and accumulating mode.
+    ReduceFnTester<Integer, Integer, IntervalWindow> tester =
+        ReduceFnTester.combining(
+            FixedWindows.of(Duration.millis(10)),
+            Never.ever(),
+            AccumulationMode.DISCARDING_FIRED_PANES,
+            new Sum.SumIntegerFn().<String>asKeyedFn(),
+            VarIntCoder.of(),
+            Duration.standardDays(365));
+
+    // Insert one element where the window it is in ends prior to the end of the global windo, but
+    // the GC time is past the end of the global window
+    tester.injectElements(
+        TimestampedValue.of(
+            0, new Instant(GlobalWindow.INSTANCE.maxTimestamp().minus(Duration.millis(15)))));
```

Comment:
Pull out constants to make this even clearer:

Duration allowedLateness = Duration.standardDays(365);
Duration windowSize = Duration.millis(10);
Instant elementTimestamp = GlobalWindow.INSTANCE.maxTimestamp().minus(allowedLateness).plus(windowSize).minus(1);

(or something like that)